File: smart_farm/src/api/errorLog.ts
import { get, http } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/** 与 smart_plant ErrorLog 字段一一对应的错误日志类型。 */
export interface ErrorLog {
  id: number
  traceId: string
  level: string
  exceptionType: string
  errorMessage?: string
  requestMethod?: string
  requestUri?: string
  operatorId?: number
  operatorName?: string
  ip?: string
  userAgent?: string
  stackTrace?: string
  handleStatus: 0 | 1 | 2
  handleRemark?: string
  handlerId?: number
  handlerName?: string
  handleTime?: string
  version: number
  createTime: string
  updateTime?: string
}

export interface ErrorLogListParams {
  traceId?: string
  operatorName?: string
  requestUri?: string
  handleStatus?: number
  startTime?: string
  endTime?: string
  pageNum: number
  pageSize: number
}

export interface ErrorLogHandlePayload {
  handleStatus: 0 | 1 | 2
  handleRemark?: string
  version: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 列表接口不返回大字段异常堆栈，保证分页查询响应稳定。 */
export const listErrorLogs = (params: ErrorLogListParams) =>
  get<ApiResult<PageInfo<ErrorLog>>>('/error-log/list', params as unknown as Record<string, unknown>)

/** 用户点击详情时再按需加载完整异常现场。 */
export const getErrorLogDetail = (id: number) =>
  get<ApiResult<ErrorLog>>(`/error-log/${id}`)

/** 修改处理状态和备注，version 用于后端乐观锁校验。 */
export const updateErrorLogHandle = (id: number, data: ErrorLogHandlePayload) =>
  http<ApiResult<ErrorLog>>({ url: `/error-log/${id}/handle`, method: 'put', data })

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/ErrorLog.java
package com.smart_plant.smart_plant.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 系统错误日志。
 * 异常现场字段只在异常发生时写入，管理员后续只能维护处理状态和备注。
 */
@Data
public class ErrorLog {

    private Long id;
    private String traceId;
    private String level;
    private String exceptionType;
    private String errorMessage;
    private String requestMethod;
    private String requestUri;
    private Long operatorId;
    private String operatorName;
    private String ip;
    private String userAgent;
    private String stackTrace;
    private Integer handleStatus;
    private String handleRemark;
    private Long handlerId;
    private String handlerName;
    private LocalDateTime handleTime;
    private Integer version;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
